qutebrowser/utils/urlutils.py

```diff
@@ -443,3 +443,60 @@ def __str__(self):
             return self.msg
         else:
             return '{}: {}'.format(self.msg, self.url.errorString())
+
+
+class IncDecError(Exception):
+
+    """Exception raised by url_incdec_number on problems.
+
+    Attributes:
+        msg: The error message.
+        url: The QUrl which caused the error.
+    """
+
+    def __init__(self, msg, url):
+        super().__init__(msg)
+        self.url = url
+        self.msg = msg
+
+    def __str__(self):
+        return '{}: {}'.format(self.msg, self.url.toString())
+
+
+def url_incdec_number(url, incdec):
+    """Find a number in the url and increment or decrement it.
+
+    Args:
+        url: The current url
+        incdec: Either 'increment' or 'decrement'
+
+    Return:
+        The new url with the number incremented/decremented.
+
+    Raises IncDecError if the url contains no number.
+    """
+    path = url.path()
+    # Get the last number in a string
+    match = re.match(r'(.*\D|^)(\d+)(.*)', path)
+    if not match:
+        raise IncDecError("No number found in URL!", url)
+    pre, number, post = match.groups()
+    if not number:
+        raise IncDecError("No number found in URL!", url)
+    try:
+        val = int(number)
+    except ValueError:
+        raise IncDecError("Could not parse number '{}'.".format(number), url)
```

Comment:
Same thing here essentially - I don't think this can ever happen with a string matching `\d+`.